provider/ansible.py

```diff
@@ -167,11 +167,15 @@ def distro_install(packages="ansible"):
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
+    ansible_install_policy = params['ansible_install_policy']
     package_list = params.get_list("package_list", 'sshpass')
     try:
         path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
-        ansible_install_policy = params['ansible_install_policy']
+        if ansible_install_policy == 'python_install':
```

Comment:
How about introducing a new param to check the ansible version? My concern is: if the system already installed an ansible pkg(rpm/dpkg), but with this change, it will still install ansible from pip. So, we can replace `path.find_command('ansible-playbook')` with `process.system(check_ansible_cmd)`. For different installation methods, just modifying the check cmd is enough.
@lkotek @yanan-fu, what do you think? Code like:
```python
if process.system(params.get("check_ansible_cmd")):
    ansible_install_policy = params['ansible_install_policy']
...
...
```

So we can remove the try block.

Reply:
Ah, for some reason there is a missing condition in try block in this snippet of code. Please check: https://github.com/autotest/tp-qemu/blob/d3e7e6551f295fb468c99eaf454b326e05e06c9b/provider/ansible.py#L174

I understand your concern and I already handle this just via L174 and checking of the installation policy. So new logic for checking ansible via pip is never used in case of rpm-installed ansible. (The whole problem is connected only with installation via pip because of previously installed avocado in development mode.)

Reply:
Let's forget our usage. If the system already installed the `ansible-core` pkg, the default policy is `distro_install`, so this case will fail. In short, We have agreed that we need the `ansible` package, but `ansible-playbook` is provided by `ansible-core`, so the current check is not enough.

However, if checking ansible package, need to consider different distros, but we can set a default one.

Just want to streamline code blocks.

Reply:
I follow, that's a good point :+1:  Please, see changes I just pushed, it reflects your idea:
```
    try:
        path.find_command('ansible-playbook')
        if ansible_install_policy == 'python_install':
            check_cmd = r"pip freeze | grep -v ansible-core | grep -q ansible="
        elif params.get("ansible_check_cmd"):
            check_cmd = params.get("ansible_check_cmd")
        if check_cmd in locals():
            LOG_JOB.debug(f"Is full ansible version installed: '{check_cmd}'")
            process.run(check_cmd, verbose=False, shell=True)
```

1. If `python_install` is used => we check using pip command
2. If `distro_install` is used and `ansible_check_cmd` is specified => we check via e.g. `rpm -q ansible` for RHEL
3. If only `distro_install` is used => we stick to the current behaviour and rely on `ansible-playbook` cmd check

I would like to keep the initial `ansible-playbook` check. It is the default behaviour a lot of users jump into and it seems it was (and still is) pretty sufficient for the most of the use cases.

Reply:
It works as expected:

`pip_install` policy + `ansible-core` installed via pip:

```
[stdlog] 2023-05-18 10:30:01,917 avocado.test DEBUG| Is full ansible version installed: 'pip freeze | grep -v ansible-core | grep -q ansible='
[stdlog] 2023-05-18 10:30:03,877 avocado.utils.process INFO | Running 'which pip'
[stdlog] 2023-05-18 10:30:03,892 avocado.utils.process DEBUG| [stdout] /usr/local/bin/pip
[stdlog] 2023-05-18 10:30:03,893 avocado.utils.process INFO | Command 'which pip' finished with 0 after 0.001811981s
[stdlog] 2023-05-18 10:30:03,894 avocado.utils.process INFO | Running 'pip install ansible'
[stdlog] 2023-05-18 10:30:05,465 avocado.utils.process DEBUG| [stdout] Collecting ansible
[stdlog] 2023-05-18 10:30:05,888 avocado.utils.process DEBUG| [stdout]   Using cached ansible-7.5.0-py3-none-any.whl (43.6 MB)
```

`distro_install` policy + `ansible-core` installed via rpm + `--customsparams="ansible_check_cmd='rpm -q ansible'"`:

```
[stdlog] 2023-05-18 10:36:30,479 avocado.test DEBUG| Is full ansible version installed: 'rpm -q ansible'
[stdlog] 2023-05-18 10:36:33,026 avocado.utils.software_manager DEBUG| dnf version: 4.14.0
[stdlog] 2023-05-18 10:36:33,027 avocado.utils.process INFO | Running 'rpm -q ansible'
[stdlog] 2023-05-18 10:36:33,046 avocado.utils.process DEBUG| [stdout] package ansible is not installed
[stdlog] 2023-05-18 10:36:33,046 avocado.utils.process INFO | Command 'rpm -q ansible' finished with 1 after 0.012010355s
[stdlog] 2023-05-18 10:36:33,047 avocado.utils.process INFO | Running '/bin/dnf -y install ansible'
```